### PR TITLE
feat: add support for exporting DID details to W3C-compliant DID Document

### PR DESCRIPTION
**Pull request by Diiaablo95**
## fixes KILTProtocol/ticket#1530

The following PR introduces a new component within the DID package: the DID Document exporter. Currently, it only supports two export formats: `application/json` and `application/json+ld`. It easily supports the addition of new mimetypes, if needed in the future.

For instructions on how to use it, please refer to the [component README](https://github.com/KILTprotocol/sdk-js/blob/144e0bff8a3228773776ac3c4a71c392c57aaf9c/packages/kilt-did/src/DidDocumentExporter/README.md) in this PR.

## How to test:

Run `yarn test` for unit tests. Integration tests have not been changed at all as this change does not affect those.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by LeonFLK** [comment]
All our sdk packages have dirs named after their package name.

Suggestion: Rename directory to match package name: /packages/kilt-did -> /packages/did

**Comment by Diiaablo95** [comment]
@LeonFLK I am fixing that in the new PR that I am gonna open soon for DID migrations 👌

File: package.json [hunk 1]
```diff
@@ -63,6 +63,6 @@
     "typedoc-plugin-external-module-name": "^4.0.6",
     "typescript": "^4.2.2"
   },
-  "version": "0.24.0-1",
+  "version": "0.24.1-rc.1",
```
Comment (tjwelde) [inline package.json:66]: Q: was this already published as this version?

**Review by tjwelde — CHANGES_REQUESTED** [review]
C: I think it would be nice to see the full DID Document somewhere. Maybe as part of the tests?

File: package.json (hunk 1 above)
Comment (Diiaablo95) [reply package.json]: https://github.com/KILTprotocol/sdk-js/commit/497f1da52fd37b7a5d6d17a561d8601ba07173b7

**Comment by Diiaablo95** [comment]
@tjwelde could not specify the exact values allowed in the export function cause that breaks the interface. I have added some comments though -> https://github.com/KILTprotocol/sdk-js/commit/506c37a0a49a508624fc39afbd612cc0204bf2a2

File: packages/kilt-did/src/DidDocumentExporter/DidDocumentExporter.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+function exportToJsonLdDidDocument(details: IDidDetails): IJsonLDDidDocument {
+  const document = exportToJsonDidDocument(details)
+  document['@context'] = ['https://www.w3.org/ns/did/v1']
+  return document as IJsonLDDidDocument
+}
+
+export function exportToDidDocument(
+  details: IDidDetails,
+  mimeType: string
```
[the remark this answers is not in the document]
Comment (Diiaablo95) [reply packages/kilt-did/src/DidDocumentExporter/DidDocumentExporter.ts:120]: https://github.com/KILTprotocol/sdk-js/commit/bf420dca151cd5dc78f56781751d0f6a5578a3fa

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**